openlibrary/macros/ShareModal.html

```diff
@@ -63,9 +62,21 @@
               <div class="share-source">$_("Copy URL")</div>
             </li>
 
+            <li>
+              <a class="qr-code-btn" title="$_('Open QR code')" data-ol-link-track="Share|QRCode">
```

Comment:
```suggestion
              <a href="/qrcode?path=$(ctx.path)" target="_blank" class="qr-code-btn" title="$_('Open QR code')" data-ol-link-track="Share|QRCode">
```

Adding an `href` with a `path` parameter will allow you to avoid adding a new JS handler for this feature.  Adding `target="_blank"` will cause the QR code link to open in a new tab.